gunicorn/config.py

```diff
@@ -20,14 +20,21 @@
 from gunicorn import __version__, util
 from gunicorn.errors import ConfigError
 from gunicorn.reloader import reloader_engines
-
-KNOWN_SETTINGS = []
+from argparse import ArgumentParser
+from gunicorn.glogging import Logger
+from gunicorn.instrument.statsd import Statsd
+from gunicorn.workers.sync import SyncWorker
+from ssl import _SSLMethod  # type: ignore[attr-defined]
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union, cast, overload
+from unittest.mock import Mock
+
+KNOWN_SETTINGS = []  # type: List[Type[Setting]]
 PLATFORM = sys.platform
 
 
-def make_settings(ignore=None):
-    settings = {}
-    ignore = ignore or ()
+def make_settings(ignore=None) -> Dict[str, "Setting"]:
+    settings = {}  # type: Dict[Any, Any]
+    ignore = ignore or ()  # type: ignore
```

Comment:
where do you define the type ignore?

Reply:
`type: ignore` tells mypy to ignore this line. I've added it in a couple of places if I didn't know directly why mypy was complaining. Those ignores should all be resolved at some point. For now, I first want to get this in general to work to get things started.

Reply:
In this specific case, very likely the problem is that the variable `ignore` first has some type, but if that value converts to bool it becomes a Tuple. Most of the time, one can resolve this by telling mypy all the possible types upfront: `type: Union[whatever it can be, Tuple]`

Reply:
fair enough. Thanks for the explanation!